app/modules/intelligence/tools/code_query_tools/git_push_tool.py

```
@@ -194,6 +194,44 @@ def _get_auth_token(self, repo_name: str) -> Optional[str]:
+        ``.git/config``. Anyone with read access to the working tree could
+        then exfiltrate a live OAuth credential.
+
+        This helper instead surfaces the failure loudly via ``logger.critical``
+        and makes a best-effort fallback attempt to overwrite the remote URL
+        directly through ``git config``, so even if ``git remote set-url``
+        fails the token does not persist locally.
+        """
+        try:
+            repo.git.remote("set-url", remote, plain_url)
+            return
+        except Exception as restore_err:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)